patches/chrome-android-java-src-org-chromium-chrome-browser-ntp-NewTabPageLayout.java.patch

```diff
@@ -0,0 +1,54 @@
+diff --git a/chrome/android/java/src/org/chromium/chrome/browser/ntp/NewTabPageLayout.java b/chrome/android/java/src/org/chromium/chrome/browser/ntp/NewTabPageLayout.java
+index db1db4f0b941c6cea83f2ca28ddc70b8e7916a90..01939137e47c16389c1ffedbc626c75868d7bec7 100644
+--- a/chrome/android/java/src/org/chromium/chrome/browser/ntp/NewTabPageLayout.java
++++ b/chrome/android/java/src/org/chromium/chrome/browser/ntp/NewTabPageLayout.java
+@@ -20,6 +20,7 @@ import android.view.ViewGroup;
+ import android.view.ViewStub;
+ import android.widget.ImageView;
+ import android.widget.LinearLayout;
++import android.widget.FrameLayout;
+ import android.widget.TextView;
+ 
+ import androidx.annotation.Nullable;
+@@ -62,7 +63,7 @@ import org.chromium.ui.base.DeviceFormFactor;
+  * Layout for the new tab page. This positions the page elements in the correct vertical positions.
+  * There are no separate phone and tablet UIs; this layout adapts based on the available space.
+  */
+-public class NewTabPageLayout extends LinearLayout implements TileGroup.Observer, VrModeObserver {
++public class NewTabPageLayout extends FrameLayout implements TileGroup.Observer, VrModeObserver {
+     private static final String TAG = "NewTabPageLayout";
+     // Used to signify the cached resource value is unset.
+     private static final int UNSET_RESOURCE_FLAG = -1;
+@@ -415,8 +416,9 @@ public class NewTabPageLayout extends LinearLayout implements TileGroup.Observer
+                 1f);
+     }
+ 
+-    private void insertSiteSectionView() {
+-        mSiteSectionView = SiteSection.inflateSiteSection(this);
++    public void insertSiteSectionView() {
```

Comment:
why is this changed to public?

Reply:
it actually makes sense to me to change this to public and then override in `BraveNewTabPageLayout`, but instead it's changed to public and then code is inserted into it? I think what you actually want to do here is just change this to public and add this code before calling `super. insertSiteSectionView()`?

Reply:
@bridiver i have tried the same thing and that's why we have public here. in this method, we add layout dynamically and we can just add layout once so instaed of creating new function, i thought of updating the same function.